src/redteam/index.ts

```diff
@@ -802,28 +802,77 @@ export async function synthesize({
 
     if (action) {
       logger.debug(`Generating tests for ${plugin.id}...`);
-      let pluginTests = await action({
-        provider: redteamProvider,
-        purpose,
-        injectVar,
-        n: plugin.numTests,
-        delayMs: delay || 0,
-        config: {
-          language,
-          modifiers: {
-            ...(testGenerationInstructions ? { testGenerationInstructions } : {}),
-            ...(plugin.config?.modifiers || {}),
+
+      // Check if plugin has its own language config override
+      const pluginLanguage = plugin.config?.language;
+
+      // If plugin has its own language, use that; otherwise use global language
+      const languages = pluginLanguage
+        ? Array.isArray(pluginLanguage)
+          ? pluginLanguage
+          : [pluginLanguage]
+        : Array.isArray(language)
+          ? language
+          : language
+            ? [language]
+            : [undefined];
```

Comment:
```suggestion
      // If plugin has its own language, use that; otherwise use global language
      const langaugeConfig = pluginLanguage ?? language;
      const langauges = Array.isArray(languageConfig) ? languageConfig : [languageConfig]
```